mesonbuild/backend/backends.py

```diff
@@ -756,17 +770,34 @@ def rpaths_for_non_system_absolute_shared_libraries(self, target: build.BuildTar
                 if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
                     continue
 
-                try:
-                    commonpath = os.path.commonpath((libdir, srcdir))
-                except ValueError: # when paths are on different drives on Windows
-                    commonpath = ''
+                if mesonlib.is_windows() and libpath.endswith('.lib'):
+
+                    if isinstance(dep, dependencies.PkgConfigDependency):
+                        # If by chance pkg-config knows the bin dir...
+                        # Hoping conan will add it to its generated pc files.
+                        found = False
+                        for index in count(1):
```

Comment:
What is this count?

Reply:
This is the convention used by conan pkgconfig generator. Since there could be more than one bindir in a conan package, it allows conan to export an arbitrary number of values for the variable. This is what conan does for `libdir` and `includedir`, so I extrapolated a little bit. (See linked conan issue...)

Reply:
Hmm, so this code is specific to a not-merged-yet conan PR? Seems premature to merge this in meson, let's see if they actually merge that PR.

That said, checking for `bindir` without number could already be done now.